docs/clipboard-unsanitized/explainer.md

```diff
@@ -0,0 +1,257 @@
+
+### In Chromium & FF:
+During `setData` call, the HTML string is written without sanitization i.e. we don’t remove tags such as `<meta>, <script>, <style>` etc from the HTML markup provided in the `setData`.
+In Chromium, the header of the HTML is hardcoded([`ui::clipboard_util::HtmlToCFHtml`](https://source.chromium.org/chromium/chromium/src/+/main:ui/base/clipboard/clipboard_util_win.cc;drc=9cc9ba08c27cb1172fb4a876ceb432f72bebfe72;l=845)) and then written to the clipboard.
+
+## Copy HTML text using async clipboard write
+```
+Version:0.9
+StartHTML:0000000105
+EndHTML:0000000252
+StartFragment:0000000141
+EndFragment:0000000216
+<html>
+<body>
+<!--StartFragment--><p style="color: red; font-style: oblique;">This text was copied </p><!--EndFragment-->
+</body>
+</html>
+
+```
+Async clipboard writer API uses sanitizers to strip out content such as `<meta>, <style>, <script>` tags etc  from the HTML. This creates issues as it’s not interop with DataTransfer’s `setData` API so web authors that use `setData` (to write HTML) and async clipboard api (to write other formats) don’t get the same content compared to using just the async clipboard write APIs for both HTML and other formats.
+
+## Copy HTML text using copy command
+```
+Version:0.9
+StartHTML:0000000170
+EndHTML:0000000770
+StartFragment:0000000206
+EndFragment:0000000734
+SourceURL:file:///C:/Users/snianu.REDMOND/Downloads/index0.html
+<html>
+<body>
+<!--StartFragment--><span style="color: rgb(0, 0, 0); font-family: &quot;Times New Roman&quot;; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;">Some text</span><!--EndFragment-->
+</body>
+</html>
+
+```
+
+Here the clipboard content is sanitized and tags such as `<meta>, <script>, <style>` etc are not included while copying contents to the clipboard.
+Another issue for HTML write: If web authors want to write the entire HTML markup using async clipboard write, then currently in Chromium it fails silently. E.g.
+```js
+async () => {
+try {
+      const html_text = new Blob([
+                  '<html><head><meta http-equiv=Content-Type content=\"text/html; charset=utf-8\"><meta name=ProgId content=Excel.Sheet><meta name=Generator content=\"Microsoft Excel 15\"><style>display:none</style></head></html>'], {type: 'text/html'});
+navigator.clipboard.write([
+                new ClipboardItem({
+                    "text/html": html_text
+                }),
+]);
+} catch(e) {
+}
+}
+
+```
+
+## Proposal
```

Comment:
We should change this to look more like a spec, ex. https://w3c.github.io/clipboard-apis/#async-clipboard-api. Proposal is fine for an explainer, but given that we can use this document as an effective spec to understand the Chromium behavior, we should structure it like a specification.

Reply:
I didn't want to duplicate all the algorithms from the clipboard API spec here as this is a minor change in `read()` & `write()` methods, and having duplicate content would most certainly result in docs going out of sync and increase confusion among web developers and UAs.
However, since this isn't an official document for clipboard APIs and has just the implementation in Chromium browser(Firefox & Safari were opposed to this feature), I can mention the IDL changes and add links to the clipboard spec for the algorithms that we are using in this feature. Thoughts?